pkgs/by-name/wa/wa-crypt-tools/package.nix

```
@@ -0,0 +1,45 @@
+{
+  python3Packages,
+  lib,
+  fetchPypi,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "wa-crypt-tools";
+  version = "0.1.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    pname = "wa_crypt_tools";
+    inherit version;
+    hash = "sha256-E/k2lKxlBxUDnfbeELHBi3sML3T+BQJ3zPHh4hGviJ4=";
+  };
+
+  build-system = [
+    python3Packages.setuptools-scm
+  ];
+
+  dependencies = [
+    python3Packages.javaobj-py3
+    python3Packages.pycryptodomex
+    python3Packages.protobuf
+    python3Packages.setuptools-scm
+  ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "protobuf >= 5.27.3,< 5.28.0" "protobuf"
+  '';
+
+  postInstallCheck = ''
+    $out/bin/wadecrypt -h
+  '';
+
+  meta = with lib; {
+    description = "Manage WhatsApp .crypt12, .crypt14 and .crypt15 files.";
+    mainProgram = "wadecrypt";
+    homepage = "https://github.com/ElDavoo/wa-crypt-tools";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ kmein ];
+  };
```

```suggestion
  meta = {
    description = "Manage WhatsApp .crypt12, .crypt14 and .crypt15 files";
    mainProgram = "wadecrypt";
    homepage = "https://github.com/ElDavoo/wa-crypt-tools";
    license = lib.licenses.gpl3Plus;
    maintainers = with lib.maintainers; [ kmein ];
  };
```
1. `with lib` before attrsets was abused to death in the past and is now frowned upon (`with` before lists is fine)
2. no punctuation at the end of `meta.description` (`meta.longDescription` is fine) 
3. license is gpl3Plus: https://github.com/ElDavoo/wa-crypt-tools/blob/0c92577e3b40a5468ab49551bec84ea3adafcd33/pyproject.toml#L23